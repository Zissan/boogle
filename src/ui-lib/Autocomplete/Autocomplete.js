import { withAutomcomplete } from "./HOC/withAutocomplete";
import TextInput from "ui-lib/TextInput";

export default withAutomcomplete(TextInput);
