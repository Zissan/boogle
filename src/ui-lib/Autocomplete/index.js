import "./Autocomplete.css";
export { default } from "./Autocomplete";
