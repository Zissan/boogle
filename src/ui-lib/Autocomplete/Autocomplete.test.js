import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Autocomplete from "./Autocomplete";

let onSelectSpy = null;
let selected;
function renderAutocomplete(args) {
  const defaultProps = {
    onChange: () => {},
    onSelect: (item) => {
      selected = item;
    },
    name: "txtAutocomplete",
    htmlId: "txtAutocomplete",
  };

  const props = { ...defaultProps, ...args };
  onSelectSpy = jest.spyOn(props, "onSelect");
  return render(<Autocomplete {...props} />);
}

it("Should render results, with two ids, autocomplete-0, autocomplete-1, in the Snapshot.", () => {
  const tree = renderAutocomplete({ items: [1, 2] });

  expect(tree).toMatchSnapshot();
});

it("Should render results, with ids, autocomplete-0, in the Snapshot.", () => {
  const { findByTestId } = renderAutocomplete({ items: [1, 2] });
  findByTestId("autocomplete-0").then((resultElement) => {
    expect(resultElement.id).toBe("autocomplete-0");
  });
});

it("Should invoke onSelect with item value 1 on clicking the result with id, autocomplete-0", () => {
  const { findByTestId } = renderAutocomplete({ items: [1, 2] });
  findByTestId("autocomplete-0").then((resultElement) => {
    fireEvent(
      resultElement,
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    );

    expect(onSelectSpy).toHaveBeenCalled();
    expect(selected).toBe(1);
  });
});
