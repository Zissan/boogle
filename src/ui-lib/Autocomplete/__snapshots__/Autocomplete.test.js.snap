// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render results, with two ids, autocomplete-0, autocomplete-1, in the Snapshot. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="autocomplete__clear__text"
      >
        X
      </span>
      <input
        aria-autocomplete="list"
        aria-expanded="true"
        autocomplete="off"
        id="txtAutocomplete"
        name="txtAutocomplete"
        role="combobox"
        type="text"
        value=""
      />
      <div
        class="autocomplete__results"
        tabindex="0"
      >
        <div
          class="autocomplete__result"
          id="autocomplete-0"
          tabindex="0"
        >
          1
        </div>
        <div
          class="autocomplete__result"
          id="autocomplete-1"
          tabindex="0"
        >
          2
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <span
      class="autocomplete__clear__text"
    >
      X
    </span>
    <input
      aria-autocomplete="list"
      aria-expanded="true"
      autocomplete="off"
      id="txtAutocomplete"
      name="txtAutocomplete"
      role="combobox"
      type="text"
      value=""
    />
    <div
      class="autocomplete__results"
      tabindex="0"
    >
      <div
        class="autocomplete__result"
        id="autocomplete-0"
        tabindex="0"
      >
        1
      </div>
      <div
        class="autocomplete__result"
        id="autocomplete-1"
        tabindex="0"
      >
        2
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
